### PR TITLE
Add integration test for adding constraints demo

### LINKED ISSUE #1280: Add integration tests for 'Adding Constraints' demo
### Problem Description

As part of migrating to SDV 1.0, we are creating new demos! Since running the old demos as a workflow caused problems (took a long time and sometimes failed due to jupyter dependencies), we want to make sure that all of the processes in the new demos are made into integration tests.

### Expected behavior

- Make integration tests for all of the `SDV` processes ran in the "Adding Constraints" demo.

### Technical Details

- Put the tests in the appropriate files. If you are adding tests that run methods for a `SingleTableSynthesizer`, put it in the file for that synthesizer ie. (`tests/integration/single_table/copulas.py`)
- We don't need to be very strict on the asserts. Just add basic asserts to ensure things aren't broken. For example,
    - Assert sampled data is the right shape
    - Validate the sampled data
    - Make sure any key values in the output are present
- Test loading from a different file. You can download the file with the custom constraint [here](https://docs.sdv.dev/sdv/reference/constraint-logic/custom-logic/example-iftruethenzero#download-the-example)

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #1280

### CODE REVIEW

**Review by frances-h — APPROVED** [review]
Looks good!

File: tests/integration/single_table/test_copulas.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    assert model_path.exists()
+    assert model_path.is_file()
+    loaded_synthesizer = GaussianCopulaSynthesizer.load(model_path)
+
+    assert isinstance(loaded_synthesizer, GaussianCopulaSynthesizer)
+    assert loaded_synthesizer.get_info() == synthesizer.get_info()
+    assert loaded_synthesizer.metadata.to_dict() == metadata.to_dict()
+    sampled_data = loaded_synthesizer.sample(100)
+    validation = sampled_data[sampled_data['has_rewards']]
+    assert validation['amenities_fee'].sum() == 0.0
```
Comment (amontanez24) [inline tests/integration/single_table/test_copulas.py:83]: can we validate the sampled data with the `synthesizer.validate` method?

File: tests/integration/single_table/test_copulas.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    # Re-Fit the model
+    synthesizer.fit(real_data)
+    synthetic_data_custom_constraint = synthesizer.sample(500)
+
+    # Assert
+    validation = synthetic_data_custom_constraint[synthetic_data_custom_constraint['has_rewards']]
+    assert validation['amenities_fee'].sum() == 0.0
+
+    # Save and Load
+    temp_dir = TemporaryDirectory()
```
Comment (amontanez24) [inline tests/integration/single_table/test_copulas.py:69]: minor: but I see in some places we use `temp_dir = TemporaryDirectory()` and in other tests we pass the `tempdir` as a parameter on the test. Is there a reason to use one over the other and if not should we unify these and just use one throughout our tests?

**Review by amontanez24 — APPROVED** [review]
A couple minor comments but otherwise LGTM!

File: tests/integration/single_table/test_copulas.py (hunk 2 above)
Comment (pvk-developer) [reply tests/integration/single_table/test_copulas.py:69]: `tempdir` as a parameter is a fixture that comes with `pytest`. I think it's pretty much the same on how we would like to proceed with this. Only pro I see is one less import but a little bit confusing for newcomers if they don't know where the `tmpdir` came from, other than that there is no strong preferences.

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**